### PR TITLE
Parameterize ChunkAggregator

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: server/src/test/scala/org/http4s/server/middleware/ChunkAggregatorSpec.scala [hunk 1]
```diff
@@ -47,7 +48,7 @@ class ChunkAggregatorSpec extends Http4sSpec {
 
     "handle a none" in {
       val routes: HttpRoutes[IO] = HttpRoutes.empty
-      ChunkAggregator(routes).run(Request()).value must returnValue(None)
+      ChunkAggregator(OptionT.liftK[IO])(routes).run(Request()).value must returnValue(None)
```
Comment (rossabaker) [inline server/src/test/scala/org/http4s/server/middleware/ChunkAggregatorSpec.scala]: Do we need some partial application trick for this one?  Ugh.

**Comment by rossabaker** [comment]
Build will be broken because of #1887.

File: server/src/main/scala/org/http4s/server/middleware/ChunkAggregator.scala [hunk 2]
```diff
@@ -2,32 +2,32 @@ package org.http4s
...
 import org.http4s.headers._
 
 object ChunkAggregator {
-  def apply[F[_]](@deprecatedName('service) routes: HttpRoutes[F])(
-      implicit F: Effect[F]): HttpRoutes[F] =
-    routes.flatMapF { response =>
-      OptionT.liftF(
+  def apply[F[_], G[_], A](f: G ~> F)(@deprecatedName('service) http: Kleisli[F, A, Response[G]])(
+      implicit F: FlatMap[F],
+      G: Sync[G]): Kleisli[F, A, Response[G]] =
```
Comment (aeons) [inline server/src/main/scala/org/http4s/server/middleware/ChunkAggregator.scala]: Same as the other one, could/should these implicits be context bounds instead.

**Comment by aeons** [comment]
I tried the partial application thing from cats, but that still crashes the compiler.
If you add the type ascription:
```scala
val nonStreamFileHttpEndpoint: HttpRoutes[F] = ChunkAggregator(OptionT.liftK[F])(fileHttpEndpoint)
```
it works as well.

**Pull request merged**